### PR TITLE
add keyboard shortcuts service

### PR DESCRIPTION
**Pull request by gzdunek**

### CODE REVIEW

File: packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcut.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import {
+  KeyboardShortcutEvent,
+  KeyboardShortcutEventSubscriber,
+  KeyboardShortcutType,
+} from './types';
+
+export function useKeyboardShortcut(
+  handler:
+    | ((shortcutEvent: KeyboardShortcutEvent) => void)
+    | Partial<Record<KeyboardShortcutType, () => void>>
```
Comment (alex-kovoy) [inline packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcut.ts:12]: Lets simply the API by allowing to subscribe only to specific bindings (without wildcards). 
```ts
type Handlers = Partial<Record<KeyboardShortcutType, () => void>>

export function useKeyboardShortcut(handlers: Handlers){

}
```

**Review by alex-kovoy — COMMENTED** [review]
Could you add a few unit-tests as well?

File: packages/teleterm/src/mainProcess/types.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ export type RuntimeSettings = {
   isDev: boolean;
   userDataDir: string;
   defaultShell: string;
+  platform: NodeJS.Platform;
```
[the remark this answers is not in the document]
Comment (gzdunek) [reply packages/teleterm/src/mainProcess/types.ts]: Good idea, done

**Comment by gzdunek** [comment]
Tests added, also simplified `KeyboardShortcutsService` by switching from `Array` to `Set` for `eventsSubscribers`.

File: packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcuts.test.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { useKeyboardShortcuts } from './useKeyboardShortcuts';
+import React from 'react';
+import AppContextProvider from 'teleterm/ui/appContextProvider';
+import { KeyboardShortcutsService } from './keyboardShortcutsService';
+import { KeyboardShortcutEventSubscriber } from './types';
+import AppContext from 'teleterm/ui/appContext';
+
+describe('teleterm/ui/services/useKeyboardShortcuts', () => {
+  it('should call handler on its event type', () => {
+    const { handler, getEventEmitter, wrapper } = getTestSetup();
```
Comment (alex-kovoy) [inline packages/teleterm/src/ui/services/keyboardShortcuts/useKeyboardShortcuts.test.tsx:11]: you do not need to use `describe` to group 1 level tests as file name and its path do the same thing (jest grouping) 

```ts
describe('teleterm/ui/services/useKeyboardShortcuts', () => {
  it('should call handler on its event type', () => {
```  

becomes

```ts
  test('handler on its event type', () => {
```  

Take a look at this guide (still in progress): 
https://gravitational.slab.com/posts/testing-guideline-proposal-wip-yuc9uvwa

**Review by alex-kovoy — APPROVED** [review]
looks good (with a minor change request)

**Pull request merged**